### PR TITLE
Create new releases

### PR DESCRIPTION
**Pull request by rail**
This new class uses a not-JSON API end-point. It was designed to be
displayed in a browser.

The idea is to use Ship-it 2 to create a task using shipitscript and submit the release data to Ship-it 1, so we can serve product details using the old infra until we port this to v2.

### CODE REVIEW

**Comment by rail** [comment]
This is how it will be implemented in shipitscript:

```python
from shipitapi import NewRelease, Release


def add_release(product, version, build_number):
    new_release = NewRelease(
        api_root="http://localhost:5000",
        auth=('admin', 'password'),
        csrf_token_prefix='{}-'.format(product),
    )
    release = Release(
        api_root="http://localhost:5000",
        auth=('admin', 'password'),
    )
    branch = "projects/maple"
    l10nChangesets = \
"""de default
it default
"""
    partials = "59.0b1build1,59.0b2build1"
    revision = "default"
    new_release.submit(
        product=product,
        version=version,
        build_number=build_number,
        branch=branch,
        revision=revision,
        l10n_changesets=l10nChangesets,
        partials=partials,
    )

    release_name = "{}-{}-build{}".format(product.capitalize(), version, build_number)
    release.update(release_name, ready=1, complete=1, status="started")


add_release("firefox", "66.0b1", 2)
```

**Comment by rail** [comment]
Actually let's simplify and move the product specific logic outside of the API. The new example usage would be:

```python 
import logging
from shipitapi import NewRelease, Release

logging.basicConfig(level=logging.DEBUG)
product = "firefox"
version = "66.0b1"
build_number = 3
branch = "projects/maple"
l10n_changesets = \
"""de default
it default
"""
partials = "59.0b1build1,59.0b2build1"
revision = "default"

data = dict(
    product=product,
    version=version,
    buildNumber=build_number,
    branch=branch,
    mozillaRevision=revision,
    l10nChangesets=l10n_changesets,
    partials=partials,
)

# Sumbmit a new release
new_release = NewRelease(
    api_root="http://localhost:5000",
    auth=('admin', 'password'),
    csrf_token_prefix='{}-'.format(product),
)
new_release.submit(**data)

# mark the release as started
release = Release(
    api_root="http://localhost:5000",
    auth=('admin', 'password'),
)
release_name = "{}-{}-build{}".format(product.capitalize(), version, build_number)
release.update(release_name, ready=1, complete=1, status="started")
```

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 setup(
     name='shipitapi',
-    version='0.1.0',
+    version='0.2.0',
```
Comment (MihaiTabara) [inline setup.py]: Even though this is non-breaking-API changes, I'd still promote a `1.0.0` as we've been using this in production for quite a while.  And now with Ship-it v2 incoming, I think it's worth an upgrade, but am fine with `0.2.0` too 😃

File: src/shipitapi/shipitapi.py [hunk 2]
```diff
@@ -108,3 +110,24 @@ def update(self, name, **data):
...
+    def submit(self, **data):
+        """Submit a new release"""
+        # Every form key should be prefixed with the product name. E.g "branch"
+        # becomes "firefox-branch".
+        product = data['product']
+        prefixed_data = {}
+        for key, value in data.items():
+            prefixed_data['{}-{}'.format(product, key)] = value
+
+        # We get a hard-to-parse HTML page. The consumers are to decide wether
```
Comment (MihaiTabara) [inline src/shipitapi/shipitapi.py]: tiny nit: s,wether,whether

**Review by MihaiTabara — APPROVED** [review]
I love this new approach, can't wait to see Ship-it v2 fully into place. This looks good to me. Feel free to send the `shipitscript` PR on my way too.

File: src/shipitapi/shipitapi.py (hunk 2 above)
Comment (rail) [reply src/shipitapi/shipitapi.py]: Bah! It's not weather though :D

**Pull request merged**